Foundation/NSFileManager.swift

```diff
@@ -343,7 +343,19 @@ open class FileManager : NSObject {
         This method replaces fileSystemAttributesAtPath:.
      */
     open func attributesOfFileSystem(forPath path: String) throws -> [FileAttributeKey : Any] {
-        NSUnimplemented()
+        var s = statfs()
+        guard statfs(path, &s) == 0 else {
+            throw _NSErrorWithErrno(errno, reading: true, path: path)
+        }
+        
+        var result = [FileAttributeKey : Any]()
+        let blockSize = UInt64(s.f_bsize)
+        result[.systemSize] = NSNumber(value: blockSize * s.f_blocks)
+        result[.systemFreeSize] = NSNumber(value: blockSize * s.f_bavail)
+        result[.systemNodes] = NSNumber(value: s.f_files)
+        result[.systemFreeNodes] = NSNumber(value: s.f_ffree)
```

Comment:
`.systemNumber` should be the `s.f_fsid.val.0`
`.systemNodes` and `.systemFreeNodes` needs `quotactl` which I am not certain is portable.